assets/case/reschedule_finish_hearing_deadline.js

```diff
@@ -1,22 +1,32 @@
 /* global $ */
 
-$('#finish_hearing_submit_button').on('click', function (e) {
+$(() => {
```

Comment:
This is now so close to be generalised. Maybe the fetching of the form should be by a class that is added to the form element?

Reply:
I agree. We could actually do it automatically for all forms inside a modal (`$('.modal form`).each(…)`), but it may also make sense to be able to do it for forms outside of modals.

I prefer to use `data-` attributes rather than the `class` attribute, e.g.

```html
<form data-use-ajax="true" …/>
```

Reply:
The `data-` attribute makes more sense.

The functionality would be nice to attach to forms outside modals too. This idea about "self-containing" re-usable forms in really small blocks is growing on me.

Reply:
I've just realised that the JS code actually looks for a modal (`const $modal = $form.closest('.modal')`) so we can, for now, only generalize for forms inside modals.

Reply:
Nice work @rimi-itk and @jekuaitk